tests/densify/test_interface_patchmatchnet.py

```diff
@@ -0,0 +1,149 @@
+"""Unit tests for the interface for PatchmatchNet
+
+In this unit test, we make a simple scenario that eight cameras are around a circle (radius = 40.0).
+Every camera's pose is towards the center of the circle (0, 0, 0). 100 track points are set on the camera plane or 
+upper the camera plane.
+
+Authors: Ren Liu
+"""
+import unittest
+
+import numpy as np
+from gtsam import PinholeCameraCal3_S2, Cal3_S2, Point3
+from gtsam.examples import SFMdata
+
+from gtsfm.common.image import Image
+from gtsfm.common.gtsfm_data import GtsfmData, SfmTrack
+from gtsfm.densify.patchmatchnet_data import PatchmatchNetData
+
+# set the default image size as 800x600, with 3 channels
+DEFAULT_IMAGE_W = 800
+DEFAULT_IMAGE_H = 600
+DEFAULT_IMAGE_C = 3
+
+# set default track points, the coordinates are in the world frame
+DEFAULT_NUM_TRACKS = 100
+DEFAULT_TRACK_POINTS = [Point3(5, 5, float(i)) for i in range(DEFAULT_NUM_TRACKS)]
+
+# set default camera intrinsics
+DEFAULT_CAMERA_INTRINSICS = Cal3_S2(
+    fx=100.0,
+    fy=100.0,
+    s=1.0,
+    u0=DEFAULT_IMAGE_W // 2,
+    v0=DEFAULT_IMAGE_H // 2,
+)
+# set default camera poses as described in GTSAM example
+DEFAULT_CAMERA_POSES = SFMdata.createPoses(DEFAULT_CAMERA_INTRINSICS)
+# set default camera instances
+DEFAULT_CAMERAS = [
+    PinholeCameraCal3_S2(DEFAULT_CAMERA_POSES[i], DEFAULT_CAMERA_INTRINSICS) for i in range(len(DEFAULT_CAMERA_POSES))
+]
+DEFAULT_NUM_CAMERAS = len(DEFAULT_CAMERAS)
+# the number of valid images should be equal to the number of cameras (with estimated pose)
+DEFAULT_NUM_IMAGES = DEFAULT_NUM_CAMERAS
+
+# build dummy image dictionary with default image shape
+DEFAULT_DUMMY_IMAGE_DICT = {
+    i: Image(value_array=np.zeros([DEFAULT_IMAGE_H, DEFAULT_IMAGE_W, DEFAULT_IMAGE_C], dtype=int))
+    for i in range(DEFAULT_NUM_IMAGES)
+}
+
+# set camera[1] to be selected in test_get_item
+EXAMPLE_CAMERA_ID = 1
+
+
+class TestPatchmatchNetData(unittest.TestCase):
+    """Unit tests for the interface for PatchmatchNet."""
+
+    def setUp(self) -> None:
+        """Set up the image dictionary and gtsfm result for the test."""
+        super().setUp()
+
+        # set the number of images as the default number
+        self._num_images = DEFAULT_NUM_IMAGES
+
+        # set up the default image dictionary
+        self._img_dict = DEFAULT_DUMMY_IMAGE_DICT
+
+        # initialize an empty sfm result in GtsfmData class
+        self._sfm_result = GtsfmData(self._num_images)
+        # add default cameras to the sfm result
+        self.add_default_cameras()
+        # add default tracks to the sfm result
+        self.add_default_tracks()
```

Comment:
self._sfm_result = get_dummy_gtsfm_data()

def  get_dummy_gtsfm_data() -> GtsfmData:
     """ """
     sfm_result = GtsfmData(self._num_images)
     # add toy data for cameras
     for i in range(DEFAULT_NUM_CAMERAS):
          self._sfm_result.add_camera(i, DEFAULT_CAMERAS[i])

     # add toy data for tracks:
        for j in range(DEFAULT_NUM_TRACKS):
            world_x = DEFAULT_TRACK_POINTS[j]
            track_to_add = SfmTrack(world_x)

            for i in range(self._num_valid_cameras):
                measurement = self._sfm_result.get_camera(i).project(world_x)
                track_to_add.add_measurement(idx=i, m=measurement)
            self._sfm_result.add_track(track_to_add)

    return sfm_result

Reply:
marking as resolved, thanks